test/discord.test.ts

```
@@ -293,161 +294,407 @@ describe('calling getUserId', () => {
-            .withMappingController(mappingControllerMock)
-            .withClient(client)
-            .withLogger(loggerMock)
-            .build();
-
-        const sentMessage = {
-            edit: jest.fn(async value => {
-                sentMessageValue = value;
-            })
-        };
-
-        const message = {
-            content: '!tasks due',
-            author: 'author',
-            channel: {
-                send: jest.fn(async () => sentMessage),
-                startTyping: jest.fn(() => Promise.resolve()),
-                stopTyping: jest.fn(() => Promise.resolve()),
-                type: 'text',
-                name: channelName
-            }
-        };
-
-        client.emit('message', message);
+    describe('when message is "!tasks in <list>"', () => {
+        it('should call commandController.taskInListForProject with list from '
+        +   'command and project ID from channel', done => {
+            const client = setupClient();
+
+            const taskList = 'Selected for Development';
+            const projectId = 0;
+            const returnedTasks = new RichEmbed();
+
+            const commandControllerMock = new CommandControllerMockBuilder()
+                .withTasksInListForProject(jest.fn(() => Promise.resolve(returnedTasks)))
+                .build();
+
+            const mappingControllerMock = new MappingControllerMockBuilder()
+                .withGetProjectId(jest.fn().mockReturnValue(projectId))
+                .build();
+
+            const discordController = new DiscordControllerBuilder()
+                .withClient(client)
+                .withMappingController(mappingControllerMock)
+                .withCommandController(commandControllerMock)
+                .build();
+
+            const message = new MessageBuilder()
+                .withContent(`!tasks in ${taskList}`)
+                .withSend(jest.fn(async value => {
+                    if (value !== returnedTasks) {
```

The test should probably fail here if the callback was called with something unexpected.